app/src/main/java/io/left/ripple/MainActivity.java

```diff
@@ -23,17 +24,17 @@
  * background colour of the peers that transmit the message along the way.
  */
 public class MainActivity extends AppCompatActivity {
-    private static final String TAG = MainActivity.class.getCanonicalName();
-
     MainViewModel viewModel;
 
     // Responsible for allowing the user to select the ping recipient.
+    @SuppressFBWarnings("UWF_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR")
     CustomViewRightMeshRecipient recipientView;
     FloatingActionButton fabSend;
     FloatingActionButton fabSendAll;
     Button buttonRed;
     Button buttonGreen;
     Button buttonBlue;
+    @SuppressFBWarnings("UWF_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR")
```

Comment:
What are the causes of these warnings, in practice?

Reply:
These warnings happen when `Findbugs` finds fields which don't call their constructor at anytime in the program.
In that case, View Objects are created by `findviewbyid` but Findbugs cannot recognize that.